### PR TITLE
feat(chip): add basic implementation of chip component

### PR DESCRIPTION
**Pull request by hcafaq**
added basic implementation of chip component with choice and filter options

### Description

image

**Updated version** 
image

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change

### How do I test this

`import Chip from '@heycar-uikit/chip';
`

### Did you remember to take care of the following?

- [x] `npm i` – for new NPM dependencies.
- [x] `npm run lint` - to check for linting issues
- [ ] `npm run test` - to run unit tests
- [ ] `npm run test:screenshots` - to run snapshot tests

### New Feature / Bug Fix

- [ ] Run unit tests to ensure all existing tests are still passing.
- [ ] Add new passing unit tests to cover the code introduced by your pr.

Thanks for contributing!

### CODE REVIEW

**Comment by hcafaq** [comment]
The **problem** it fails on
```
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
@heycar-uikit/icons/Close (imported by src/Chip.tsx)
[!] (plugin Typescript) TS2307: Cannot find module '@heycar-uikit/icons/Close' or its corresponding type declarations.
src/Chip.tsx (5:23)

5 import CloseIcon from '@heycar-uikit/icons/Close';
```
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
**dependency in the package.json**
```
"dependencies": {
    "classnames": "^2.3.1",
    "@heycar-uikit/icons": "1.3.1"
  },
```
the import
`import CloseIcon from '@heycar-uikit/icons/Close';
`




**Suggested fixes**
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
https://www.npmjs.com/package/@rollup/plugin-node-resolve

@derek-heycar @emrahsemizheycar

**Comment by derek-heycar** [comment]
@hcafaq Several points here

* provides to pass all of the attributes to the component relating to the general HTML element to the component.
* do not forget about SH tests.
* Please double-check the naming of the CSS classes for the element, (sometimes better to use CamelCase), just check different components.
* and pls, to be attentive.

**Comment by hcafaq** [comment]
Updated design 
image

File: packages/chip/src/Chip.types.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { AnchorHTMLAttributes } from 'react';
+
+export interface ComponentProps {
+  /**
+   * Customize Chip text color
+   */
+  variant?: 'filter' | 'choice';
```
Comment (derek-heycar) [inline packages/chip/src/Chip.types.ts:7]: That should be decided by the `onDelete` callback

File: packages/chip/src/Chip.types.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { AnchorHTMLAttributes } from 'react';
+
+export interface ComponentProps {
```
Comment (derek-heycar) [inline packages/chip/src/Chip.types.ts:3]: Where is `onDelete` cb ?

File: packages/chip/src/Chip.types.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  /**
+   * Customize Chip text color
+   */
+  variant?: 'filter' | 'choice';
+  /**
+   * The content of the Chip
+   */
+  children: React.ReactNode;
+  /**
+   * 	boolean to make chip disable.
```
Comment (derek-heycar) [inline packages/chip/src/Chip.types.ts:13]: extra spaces, formate/lint?

File: packages/chip/src/Chip.types.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+   */
+  disabled?: boolean;
+  /**
+   * selected style for choice chip
+   */
+  selected?: boolean;
```
Comment (derek-heycar) [inline packages/chip/src/Chip.types.ts:19]: Hmmm. it looks like something strange.
how it should work ? 
we have two events on one component, click on delete icon and click on chip. do you have any particular cases for that ?

File: packages/chip/src/Chip.types.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  /**
+   * The component used for the root node. Either a string to use a HTML element or a component
+   */
+  Component?: React.ElementType;
+  /**
+   * Element placed after the children.
+   */
+  leftIcon?: React.ReactNode;
+  /**
+   * 	Element before after the children.
```
Comment (derek-heycar) [inline packages/chip/src/Chip.types.ts:29]: space

File: packages/chip/src/__tests__/Chip.screenshots.test.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import {
+  createStorybookUrl,
+  setupScreenshotTesting,
+} from '../../../screenshotUtils';
+
+const packageName = 'chip';
+const chipLabel = 'Chip CTA';
+const chipVariants = ['filter', 'choice'];
+const clip = { x: 0, y: 0, width: 200, height: 100 };
```
Comment (derek-heycar) [inline packages/chip/src/__tests__/Chip.screenshots.test.tsx:9]: where is sh images ?

File: packages/chip/src/__tests__/Chip.screenshots.test.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        packageName,
+        knobs: {
+          variant,
+          children: chipLabel,
+        },
+      }),
+    ]),
+    screenshotOpts: { clip },
+  }),
+);
```
Comment (derek-heycar) [inline packages/chip/src/__tests__/Chip.screenshots.test.tsx:32]: Where is checks for all of behaves of the component  Default Hover Pressed Disable?

File: packages/chip/src/__tests__/Chip.snapshot.test.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    ).toMatchSnapshot();
+  });
+
+  it('should render selected="true"', () => {
+    expect(
+      render(<Chip selected={true}>{defaultChild}</Chip>),
+    ).toMatchSnapshot();
+  });
+
+  it('should render Component="span"', () => {
```
Comment (derek-heycar) [inline packages/chip/src/__tests__/Chip.snapshot.test.tsx:37]: better to add all API cases

File: packages/chip/src/__tests__/Chip.test.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+      expect(container.firstElementChild).toHaveClass('choice');
+    });
+
+    it('should be selected ', () => {
+      const { container } = render(
+        <Chip selected={true} variant="choice">
+          {defaultChild}{' '}
+        </Chip>,
+      );
+
```
Comment (derek-heycar) [inline packages/chip/src/__tests__/Chip.test.tsx:73]: copy/past spaces

**Review by derek-heycar — CHANGES_REQUESTED** [review]

File: packages/chip/src/Chip.types.ts (hunk 1 above)
Comment (hcafaq) [reply packages/chip/src/Chip.types.ts:7]: filter can be without onDelete too, so this is needed

image

File: packages/chip/src/Chip.types.ts (hunk 2 above)
Comment (hcafaq) [reply packages/chip/src/Chip.types.ts:3]: resolved

File: packages/chip/src/Chip.types.ts (hunk 4 above)
Comment (hcafaq) [reply packages/chip/src/Chip.types.ts:27]: both cases are handled in stories.

Comment (hcafaq) [reply packages/chip/src/Chip.types.ts:27]: resolved

File: packages/chip/src/__tests__/Chip.test.tsx (hunk 9 above)
Comment (hcafaq) [reply packages/chip/src/__tests__/Chip.test.tsx:73]: running formatter prettier or eslint has not effect on this one, it is properly formatted

File: packages/chip/src/__tests__/Chip.snapshot.test.tsx (hunk 8 above)
Comment (hcafaq) [reply packages/chip/src/__tests__/Chip.snapshot.test.tsx:37]: resolved

File: packages/chip/src/__tests__/Chip.screenshots.test.tsx (hunk 6 above)
Comment (hcafaq) [reply packages/chip/src/__tests__/Chip.screenshots.test.tsx:9]: updated

File: packages/chip/src/__tests__/Chip.screenshots.test.tsx (hunk 7 above)
Comment (hcafaq) [reply packages/chip/src/__tests__/Chip.screenshots.test.tsx:34]: added

File: packages/chip/src/Chip.types.ts (hunk 1 above)
Comment (derek-heycar) [reply packages/chip/src/Chip.types.ts:7]: It needs to figure out. But generally, it is not a variant param it is a color. We should keep our main naming API convension.

File: .storybook/utils/componentsResolver.js [hunk 10]
```diff
@@ -6,23 +6,94 @@ const componentsDir = path.resolve(__dirname, '../../packages');
...
+          descriptionFileData,
+          request: requestPath,
+        } = request;
+
+        if (requestPath && requestPath.match(componentsPrefix)) {
+
+          const componentName = requestPath.replace(`./${componentsPrefix}`, '');
+          const path = `${componentsDir}`;
+
+        console.log('requestPath -->', requestPath);
```
Comment (esraltintas) [inline .storybook/utils/componentsResolver.js:28]: Hi @hcafaq do we need to keep console.logs?

**Review by derek-heycar — CHANGES_REQUESTED** [review]
Could you please close the PR, till ready component-time ?

**Comment by hcafaq** [comment]
This PR remains to be updated until new design is finalized

**Comment by hcafaq** [comment]
closing this PR until new work on chips has started

**Pull request closed without merging**